Assets/Scripts/PickUps.cs

```diff
@@ -36,33 +39,38 @@ private void OnTriggerEnter2D(Collider2D other)
         if (other.tag == "Player")
         {
             PlayerController player = other.GetComponent<PlayerController>();
+            
+            if (gameObject.name == "HealthUp(Clone)")
+            {
+                if (PlayerController.currentHP < player.maxHP)
+                {
+                    PlayerController.currentHP++;
+                    Instantiate(feedbacks, transform.position, transform.rotation);
+                    SoundManager.instance.PlayAcquire();
+                    Destroy(gameObject);
+                }
+            }
+
             if (gameObject.name == "SpeedUp(Clone)")
             {
-                if (player.speed < 7)
+                if (player.speed < speedLimite)
                 {
                     player.speed += .8f;

```

Comment:
I would say to make a variable for this number as well haha, I think you get the idea but I just want to make sure you don't accidentally overlook any